ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/CompactionExecutorFactory.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.hadoop.hive.ql.txn.compactor;
+
+import org.apache.hadoop.hive.metastore.api.Table;
+import org.apache.hadoop.hive.metastore.utils.MetaStoreUtils;
+
+public class CompactionExecutorFactory {
+  
+  public static CompactionExecutor getInstance(Table table, Worker worker, Worker.CompactionTxn compactionTxn, 
```

Comment:
```` 
 private static final CompactionExecutorFactory INSTANCE = new CompactionExecutorFactory();

  public static CompactionExecutorFactory getInstance() {
    return INSTANCE;
  }
````

Reply:
the current method should be 'createExecutor(Table table)'

Reply:
Done.

I named getInstance based on other code examples. 
For example, in TaskHandlerFactory class.